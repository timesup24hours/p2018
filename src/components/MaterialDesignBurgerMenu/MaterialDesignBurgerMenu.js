import React from 'react';

import styles from './index.css';

// export default ({ CustomhandleOnClick }) => {
//   let spanRef;
//   const handleOnClick = e => {
//     if (CustomhandleOnClick) CustomhandleOnClick(e);
//     spanRef.classList.toggle(styles.trigger);
//   };
//   return (
//     <div className={styles.wrapper} onClick={handleOnClick}>
//       <div className={styles.spanWrapper}>
//         <span ref={ref => (spanRef = ref)} className={styles.stick} />
//       </div>
//     </div>
//   );
// };

export default class MaterialDesignBurgerMenu extends React.PureComponent {
  state = {
    open: false
  };

  handleOnClick = e => {
    if (this.props.CustomhandleOnClick) this.props.CustomhandleOnClick(e);
    this.spanRef.classList.toggle(styles.trigger);
  };

  // static getDerivedStateFromProps(props, state) {
  //   console.log(props, ' ', state);
  //   if (props.open !== state.open) {
  //     return {
  //       open: props.open
  //     };
  //   }
  //   return null;
  // }

  componentDidUpdate(prevProps, prevState, snapshot) {
    if (this.props.open) {
      this.spanRef.classList.add(styles.trigger);
    } else {
      this.spanRef.classList.remove(styles.trigger);
    }
    // if (!this.props.open && this.spanRef.classList.contains(styles.trigger)) {
    //   this.spanRef.classList.remove(styles.trigger);
    // } else if (
    //   this.props.open &&
    //   !this.spanRef.classList.contains(styles.trigger)
    // ) {
    //   this.spanRef.classList.add(styles.trigger);
    // }
  }

  render() {
    return (
      <div className={styles.wrapper} onClick={this.handleOnClick}>
        <div className={styles.spanWrapper}>
          <span ref={ref => (this.spanRef = ref)} className={styles.stick} />
        </div>
      </div>
    );
  }
}
