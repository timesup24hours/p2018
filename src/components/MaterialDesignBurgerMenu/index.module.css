.container {
  position: absolute;
  display: block;
  cursor: pointer;
  width: 40px;
  height: 40px;
}

.wrapper {
  height: 100%;
  width: 100%;
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  /* background: white; */
  /* border-radius: 4px; */
  /* padding-bottom: 4px; */
  /* border: 1px solid black; */
}
.spanWrapper {
  height: 80%;
  width: 80%;
  position: relative;

  /* border: 1px solid darkgreen; */
  display: flex;
  justify-content: center;
  align-items: center;
}

.stick,
.stick:before,
.stick:after {
  width: 100%;
  height: 4px;
  /* margin: 0 auto; */
  /* background: #75489f; */
  background: white;
  position: absolute;
  transition: all 0.6s;
}

.stick:before {
  content: '';
  top: -13px;
  left: 0;
  position: absolute;
}

.stick:after {
  content: '';
  top: 13px;
  left: 0;
  position: absolute;
}

.stick.trigger {
  transform: rotate(180deg);
}

.stick.trigger:after {
  transform-origin: 100% 50%;
  transform: translateY(-14px) rotate(-45deg) scaleX(0.7);
}

.stick.trigger:before {
  transform-origin: 100% 50%;
  transform: translateY(14px) rotate(45deg) scaleX(0.7);
}
