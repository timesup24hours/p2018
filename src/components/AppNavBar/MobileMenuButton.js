import React from 'react';
import styles from './index.module.css';
import { MaterialDesignBurgerMenu } from '../MaterialDesignBurgerMenu';

export default React.forwardRef((props, ref) => (
  <div className={styles.menuButton} ref={ref}>
    <MaterialDesignBurgerMenu
      customhandleOnClick={props.customhandleOnClick}
      open={props.open}
    />
  </div>
));
