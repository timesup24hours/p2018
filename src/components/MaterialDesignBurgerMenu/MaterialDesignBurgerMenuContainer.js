import React from 'react';

import styles from './index.module.css';
import MaterialDesignBurgerMenu from './MaterialDesignBurgerMenu';

export default () => {
  return (
    <div className={styles.container}>
      <MaterialDesignBurgerMenu />
    </div>
  );
};
