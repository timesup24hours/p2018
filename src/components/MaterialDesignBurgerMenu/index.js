// import React from 'react';
import MaterialDesignBurgerMenuContainer from './MaterialDesignBurgerMenuContainer';
import MaterialDesignBurgerMenu from './MaterialDesignBurgerMenu';

export { MaterialDesignBurgerMenuContainer, MaterialDesignBurgerMenu };
