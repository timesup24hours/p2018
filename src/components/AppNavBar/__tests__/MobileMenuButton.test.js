import React from 'react';
import { shallow, mount } from 'enzyme';
import MobileMenuButton from '../MobileMenuButton';
import ReactDOM from 'react-dom';
import { MaterialDesignBurgerMenu } from '../../MaterialDesignBurgerMenu';

describe('MobileMenuButton', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<MobileMenuButton debug />);

    expect(component).toMatchSnapshot();
    // console.log(component.debug());
  });

  it('passes refs to the container component', () => {
    const ref = React.createRef();
    const div = document.createElement('div');
    ReactDOM.render(<MobileMenuButton ref={ref} />, div);

    const element = div.querySelector('div');
    expect(element).toBe(ref.current);
  });

  // it('toggle method should change the state', () => {
  //   const customhandleOnClick = jest.fn();
  //   const open = true;
  //   const ref = React.createRef();
  //   const wrapper = mount(
  //     <MobileMenuButton
  //       customhandleOnClick={customhandleOnClick}
  //       open={open}
  //       ref={ref}
  //     />
  //   );
  //   const shallowWrapper = shallow(
  //     <MobileMenuButton
  //       customhandleOnClick={customhandleOnClick}
  //       open={open}
  //       ref={ref}
  //     />
  //   );

  //   const instance = wrapper.instance();
  //   console.log('instance: ', instance);
  //   console.log('mount: ', wrapper.debug());
  //   console.log('shallowWrapper: ', shallowWrapper.debug());
  //   // instance.toggle();
  //   // expect(wrapper.state().isOpen).toBeFalsy();

  //   const shallowWrapperProps = shallow(
  //     <MobileMenuButton
  //       customhandleOnClick={customhandleOnClick}
  //       open={open}
  //       ref={ref}
  //     />
  //   )
  //     .find(MaterialDesignBurgerMenu)
  //     .renderProp('customhandleOnClick');
  //   // console.log('shallowWrapperProps: ', shallowWrapperProps);
  // });

  it('pass prop function into the child component', () => {
    const customhandleOnClick = jest.fn();
    const ref = React.createRef();

    const child = shallow(
      <MobileMenuButton customhandleOnClick={customhandleOnClick} ref={ref} />
    ).find(MaterialDesignBurgerMenu);

    const childProps = child.props();
    // console.log('childProps: ', childProps);

    expect(childProps.customhandleOnClick).toBe(customhandleOnClick);
    expect(childProps.open).toBe(undefined);
  });

  it('pass prop boolean into the child component', () => {
    const open = true;
    const ref = React.createRef();

    const child = shallow(<MobileMenuButton open={open} ref={ref} />).find(
      MaterialDesignBurgerMenu
    );
    const childProps = child.props();
    expect(childProps.open).toBe(open);
    expect(childProps.customhandleOnClick).toBe(undefined);
  });
});
