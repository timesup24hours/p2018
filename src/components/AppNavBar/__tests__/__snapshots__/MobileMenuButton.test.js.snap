// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileMenuButton should render correctly in "debug" mode 1`] = `
<div
  className="menuButton"
>
  <MaterialDesignBurgerMenu />
</div>
`;
